client/telemetry/src/lib.rs

```diff
@@ -129,64 +113,61 @@ pub struct TelemetryWorker {
 	message_sender: mpsc::Sender<TelemetryMessage>,
 	register_receiver: mpsc::UnboundedReceiver<Register>,
 	register_sender: mpsc::UnboundedSender<Register>,
+	id_counter: Arc<atomic::AtomicU64>,
 	transport: WsTrans,
 }
 
 impl TelemetryWorker {
-	pub(crate) fn new(buffer_size: usize, transport: WsTrans) -> Self {
+	/// Instantiate a new [`TelemetryWorker`] which can run in background.
+	///
+	/// Only one is needed per process.
+	pub fn new(buffer_size: usize, transport: Option<ExtTransport>) -> Result<Self> {
```

Comment:
Maybe we could add a second method `with_transport` where we can pass the transport. Normally we don't need this.

Reply:
True!! Thanks